### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by sdelamo**
This PRs: 

- Adds a feature similar to the one we will [need to create in starter](https://github.com/micronaut-projects/micronaut-starter/issues/1446). 
- Use `  "base": "micronaut-data-jdbc-repository"` to generate a sample project with the starting point we tell the user. 

I have moved parts of the contents of this PR to this other: 

https://github.com/micronaut-projects/micronaut-guides/pull/1103

### CODE REVIEW

File: guides/micronaut-cloud-secrets-azure/src/main/resources/bootstrap.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  application:
+    name: micronautguide # <1>
+  config-client:
+    enabled: true # <2>
+#end::distributedconfiguration[]
+#tag::azure[]
+azure:
+  credential:
+    client-secret:
+      client-id: ${AZURE_CLIENT_ID} # <1>
```
Comment (sbodvanski) [inline guides/micronaut-cloud-secrets-azure/src/main/resources/bootstrap.yml:12]: is this going to work given the Azure naming convention restriction for secrets, which does not allow underscores to be used?  Should it be `AZURE-CLIENT-ID`?

File: guides/micronaut-cloud-secrets-azure/src/main/resources/bootstrap.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+      tenant-id: ${AZURE_TENANT_ID} # <2>
```
Comment (sbodvanski) [inline guides/micronaut-cloud-secrets-azure/src/main/resources/bootstrap.yml:13]: The same comment as above for all the `AZURE_` secret entries.

**Comment by sdelamo** [comment]
sorry, I have accidentally merged this. Let me know if you have any concerns. 

I have changed `AZURE_CLIENT_ID` to `AZURE-CLIENT-ID`. However, it was written as `AZURE_CLIENT_ID` in your original commits.

**Comment by sbodvanski** [comment]
I have no additional concerns.

**Pull request merged**